### PR TITLE
Adds RefreshSentinel type as a generic key/multi-value accumulator.

### PR DESCRIPTION
**Pull request by ghost**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/sentinel/41)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/refresh_sentinel.rs [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+/// Entry for accumulation.
+#[derive(Clone)]
+pub struct Entry<V> {
+    /// Values accumulated for a given key.
+    pub received_response: Vec<V>,
+}
+
+/// Generic type for accumulating multiple values under a given key.
+#[allow(dead_code)]
+pub struct RefreshSentinel<K, V> where K: PartialOrd + Ord + Clone, V: Clone {
```
Comment (benjaminbollen) [inline src/refresh_sentinel.rs:36]: Do we need K: PartialOrd + Ord for LruCache on Key?  I assume yes.

File: src/refresh_sentinel.rs [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    }
+
+    /// Adds a key/value pair, if the key already exists add the value under that key.
+    /// Optionally returns the key and the vector of values if the quroum has been reached. 
+    #[allow(dead_code)]
+    pub fn add(&mut self, key: K, value: V)-> Option<(K, Vec<V>)> {
+        let entry = self.storage.remove(&key);
+        if entry.is_none() {
+            let entry_in = Entry { received_response : vec![value]};
+            self.storage.add(key.clone(), entry_in.clone());
```
Comment (benjaminbollen) [inline src/refresh_sentinel.rs:74]: we don't need to clone() entry_in here, same for key - I think, Im not rustc of course - moving the value suffices;
but it requires restructuring a bit, so you can ignore my comment!

**Comment by benjaminbollen** [comment]
partially awesome ! ;-)

the only overall comment is that it maybe should not be called a "sentinel", but that's just semantics for now.  "refresh_accumulator" would be more appropriate.

**Comment by benjaminbollen** [comment]
looks good to me;

**Pull request merged**